rust/lancedb/src/index/vector.rs

```diff
@@ -53,9 +53,14 @@ pub struct VectorIndexMetadata {
 }
 
 #[derive(Debug, Deserialize)]
-pub struct VectorIndexStatistics {
+pub struct IndexMetadata {
     pub num_indexed_rows: usize,
     pub num_unindexed_rows: usize,
+}
+
+#[derive(Debug, Deserialize)]
+pub struct VectorIndexStatistics {
+    pub num_indexed_and_unindexed_rows: IndexMetadata,
```

Comment:
naming nits:

num_indexed_and_unindexed_rows is overly specific, and "Metadata" is somewhat redundant

how about:
```
row_stats: IndexRowStatistics
```